src/decimal.rs

```diff
@@ -1694,8 +1694,68 @@ const fn flags(neg: bool, scale: u32) -> u32 {
     (scale << SCALE_SHIFT) | ((neg as u32) << SIGN_SHIFT)
 }
 
+macro_rules! impl_try_from_decimal {
+    ($TInto:ty, $conversion_fn:path, $name:expr) => {
+        ///
+        /// Try to convert a `Decimal` to a primitive type.
+        ///
+        /// Can fail if the `Decimal` is out of range.
+        /// For integer types, truncate and return the integer part.
+        ///
+        impl core::convert::TryFrom<Decimal> for $TInto {
+            type Error = crate::Error;
+
+            #[inline]
+            fn try_from(t: Decimal) -> Result<Self, Error> {
+                $conversion_fn(&t).ok_or_else(|| Error::from(format!("Failed to convert to {}", $name)))
+            }
+        }
+    };
+}
+
+impl_try_from_decimal!(f32, Decimal::to_f32, "f32");
+impl_try_from_decimal!(f64, Decimal::to_f64, "f32");
+impl_try_from_decimal!(isize, Decimal::to_isize, "isize");
+impl_try_from_decimal!(i8, Decimal::to_i8, "i8");
+impl_try_from_decimal!(i16, Decimal::to_i16, "i16");
+impl_try_from_decimal!(i32, Decimal::to_i32, "i32");
+impl_try_from_decimal!(i64, Decimal::to_i64, "i64");
+impl_try_from_decimal!(i128, Decimal::to_i128, "i128");
+impl_try_from_decimal!(usize, Decimal::to_usize, "usize");
+impl_try_from_decimal!(u8, Decimal::to_u8, "u8");
+impl_try_from_decimal!(u16, Decimal::to_u16, "u16");
+impl_try_from_decimal!(u32, Decimal::to_u32, "u32");
+impl_try_from_decimal!(u64, Decimal::to_u64, "u64");
+impl_try_from_decimal!(u128, Decimal::to_u128, "u128");
+
+macro_rules! impl_try_from_primitive {
+    ($TFrom:ty, $conversion_fn:path) => {
+        ///
+        /// Conversion to `Decimal`.
```

Comment:
Consider utilizing the `doc` macro here too. e.g. 

```
#[doc = concat!(
            "Try to convert a `",
            stringify!($TFrom),
            "` into a `Decimal`."
        )]
```

It's an interesting one for erroring here - it'll only likely error on `128` bit integers. This could be again passed through as additional doc if need be.